scilpy/tracking/tracking_field.py

```diff
@@ -1,82 +1,156 @@
 # -*- coding: utf-8 -*-
-import logging
-
 import dipy.data
 from dipy.reconst.shm import sh_to_sf_matrix, order_from_ncoef
 import numpy as np
 
+from scilpy.image.datasets import DataVolume
 from scilpy.tracking.tools import sample_distribution
 from scilpy.tracking.utils import TrackingDirection
 
 
-class SphericalHarmonicField(object):
+class AbstractTrackingField(object):
     """
     Spherical harmonics tracking field.
 
     Parameters
     ----------
-    odf_dataset: scilpy Dataset
-        Trackable Dataset object.
-    basis: string
-        SH basis name. One of 'tournier07' or 'descoteaux07'
-    sf_threshold: float
-        Threshold on spherical function (SF).
-    sf_threshold_init: float
-        Threshold on spherical function when initializing a new streamline.
+    dataset: Any
+        Dataset object.
     theta: float
         Maximum angle (radians) between two steps.
-    dipy_sphere: string
-        Name of the DIPY sphere object to use for evaluating SH.
+    dipy_sphere: string, optional
+        Name of the DIPY sphere object to use for evaluating SH. Can be set to
+        None to skip sphere initialization.
     """
-
-    def __init__(self, odf_dataset, basis, sf_threshold, sf_threshold_init,
-                 theta, dipy_sphere='symmetric724'):
-        self.sf_threshold = sf_threshold
-        self.sf_threshold_init = sf_threshold_init
+    def __init__(self, dataset, theta, dipy_sphere=None):
         self.theta = theta
+        self.dataset = dataset
+
+        if dipy_sphere:
+            if 'symmetric' not in dipy_sphere:
+                raise ValueError('Sphere must be symmetric. Call to '
+                                 'get_opposite_direction will fail.')
+            self.sphere = dipy.data.get_sphere(dipy_sphere)
+            self.dirs = np.zeros(len(self.sphere.vertices), dtype=np.ndarray)
+            for i in range(len(self.sphere.vertices)):
+                self.dirs[i] = TrackingDirection(self.sphere.vertices[i], i)
+            self.tracking_neighbours = self._get_sphere_neighbours(self.theta)
+        else:
+            self.sphere = None
+            self.dirs = None
+            self.tracking_neighbours = None
+
+    def _get_sphere_neighbours(self, max_angle):
```

Comment:
In the end, it is easier from integration with dwi_ml to avoid a child class and to have this as an option.

I add a check in this method, and get_opposite_direction_sphere won't crash now if self.dirs = [] instead of None.